### PR TITLE
feat: make hard-coded authz adapter editable, rename adapter to ormer

### PR DESCRIPTION
**Pull request by leo220yuyaodog**
1. add built in adapter and model when init the database
2. make hard-code adapter and model in authz package editable and visible in Casdoor UI.
3. optimize the experience of editing adapter configration.

Now codes about adapter and enforcer are incremental update, ensuring that existing functionality is not broken.

https://github.com/casdoor/casdoor/assets/71440988/2df131d1-55b4-4cf1-a942-fdbbbe37712d

### CODE REVIEW

File: web/src/Setting.js [hunk 1]
```diff
@@ -1218,3 +1218,19 @@ export function isDefaultOrganizationSelected(account) {
...
+const BuiltInResources = [
+  "authz-enforcer-built-in",
+  "permission-enforcer-built-in",
+  "authz-model-built-in",
+  "permission-model-built-in",
+  "authz-adapter-built-in",
+  "permission-adapter-built-in",
+];
+
+export function buildInResource(obj) {
```
Comment (hsluoyz) [inline web/src/Setting.js:1231]: isBuiltInObject

File: web/src/Setting.js [hunk 2]
```diff
@@ -1218,3 +1218,19 @@ export function isDefaultOrganizationSelected(account) {
   }
   return false;
 }
+
+const BuiltInResources = [
+  "authz-enforcer-built-in",
```
Comment (hsluoyz) [inline web/src/Setting.js:1223]: All `authz` to `api`

File: web/src/ModelEditPage.js [hunk 3]
```diff
@@ -91,6 +91,7 @@ class ModelEditPage extends React.Component {
   }
 
   renderModel() {
+    const buildInResource = Setting.buildInResource(this.state.model);
```
Comment (hsluoyz) [inline web/src/ModelEditPage.js:94]: Don't define var

File: web/src/EnforcerEditPage.js [hunk 4]
```diff
@@ -108,6 +108,7 @@ class EnforcerEditPage extends React.Component {
   }
 
   renderEnforcer() {
+    const buildInResource = Setting.buildInResource(this.state.enforcer);
```
Comment (hsluoyz) [inline web/src/EnforcerEditPage.js:111]: Don't define var

File: web/src/AdapterListPage.js [hunk 5]
```diff
@@ -32,7 +32,7 @@ class AdapterListPage extends BaseListPage {
       createdTime: moment().format(),
       type: "Database",
       host: "localhost",
-      port: 3306,
+      port: "3306",
```
Comment (hsluoyz) [inline web/src/AdapterListPage.js:35]: Isn't port int type?

File: web/src/AdapterEditPage.js [hunk 6]
```diff
@@ -104,7 +104,75 @@ class AdapterEditPage extends React.Component {
...
+              </Row>
+            </React.Fragment>
+          )
+      }
+      </React.Fragment>
+    );
+  }
+
   renderAdapter() {
+    const buildInResource = Setting.buildInResource(this.state.adapter);
```
Comment (hsluoyz) [inline web/src/AdapterEditPage.js:175]: Don't define var

File: object/init.go [hunk 7]
```diff
@@ -358,3 +414,109 @@ func initBuiltInPermission() {
...
+		IsEnabled:   true,
+	}
+
+	_, err = AddEnforcer(permissionEnforcer)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func initBuiltInAuthzEnforcer() {
```
Comment (hsluoyz) [inline object/init.go:498]: Authz -> Api

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@leo220yuyaodog change all ORM's `Adatper` to `Ormer`, let Casbin use `Adapter`

File: web/src/AdapterListPage.js (hunk 5 above)
Comment (leo220yuyaodog) [reply web/src/AdapterListPage.js:35]: Has change to string type, it is more flexiable than int.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**